### DISCUSSION TITLE
Version confusion

### DISCUSSION THREAD

**Discussion by RFC1920** [jchristn/SuperSimpleTcp#65] [Ideas]
FYI, the name of this project matches exactly the WRONG Nuget package.  So, I search that name and found the other one.  Similar but not the same.

Fortunately, having installed SuperSimpleTcp instead, all appears to be working.

**Comment by jchristn** [comment]
Yeah, someone beat me to the punch on SimpleTcp.

Everything is working ok with my package?

**Comment by jchristn** [comment]
By the way, hats off to you for having a reference to Jon Postel's work in your github username :)  That guy was a legend.

**Comment by RFC1920** [comment]
So far so good.   Thanks!  I honestly wanted to use Watson but was running into that painful Newtonsoft conflict.  After tearing things down and building them back up I might still have a chance.   But, this library should suffice.

**Comment by jchristn** [comment]
You could implement your own serializer methods and that would do it.  I was thinking of adding custom serializer support in a future release.  

Example in WatsonMessage.cs
```csharp
        /// <summary>
        /// Message headers in byte-array form ready to send.
        /// </summary>
        [JsonIgnore]
        public byte[] HeaderBytes
        {
            get
            {
                string jsonStr = SerializationHelper.SerializeJson(this, false);
                byte[] jsonBytes = Encoding.UTF8.GetBytes(jsonStr);
                byte[] end = Encoding.UTF8.GetBytes("\r\n\r\n");
                byte[] final = WatsonCommon.AppendBytes(jsonBytes, end);
                return final;
            }
        }
```

If you change the ```SerializationHelper``` methods (you can use BSON, XML, whatever you like, as long as newline are encoded) then it should work
